installer/client/src/core/util/composable/useProgress.ts

```diff
@@ -0,0 +1,67 @@
+/**
+ * OrangeHRM is a comprehensive Human Resource Management (HRM) System that captures
+ * all the essential functionalities required for any enterprise.
+ * Copyright (C) 2006 OrangeHRM Inc., http://www.orangehrm.com
+ *
+ * OrangeHRM is free software; you can redistribute it and/or modify it under the terms of
+ * the GNU General Public License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * OrangeHRM is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program;
+ * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA  02110-1301, USA
+ */
+
+import {ref} from 'vue';
+
+export default function useProgress() {
+  let time = 0;
+  let working = true;
+  let generator: Generator<number>;
+  const frequency = 100;
+  const progress = ref(0);
+
+  function* exponentGenerator() {
```

Comment:
![image](https://user-images.githubusercontent.com/12880059/164312701-59932229-f150-4140-bcfc-0a96f4016bc4.png)
Following this explanation from [fakeprogress](https://github.com/piercus/fake-progress#fakeprogress) our implementation was done using generator function. can tweak more with frequency values if needed.